app/models/infrastructure_facades/clouds/abstract_cloud_client.rb

```diff
@@ -2,6 +2,7 @@
 # - all_images_info -> get array of hashes: image_identifier => image label for all images permitted to use by cloud user
 # - instantiate_vms(base_instace_name, image_id, number) => list of instance ids (Strings)
 # - all_vm_ids -> get array of VM ids (Strings)
+# - get_subinfrastructures(user_id) -> list of hashes representing distinct types of infrastructure
```

Comment:
types -> configurations
wszędzie, chyba, że masz argumenty, że to złe określenie i jest lepsze